pkg/api/types.go

```diff
@@ -2103,6 +2103,16 @@ type PodSpec struct {
 	// file if specified. This is only valid for non-hostNetwork pods.
 	// +optional
 	HostAliases []HostAlias
+	// If specified, indicates the pod's priority. "system" is a special keyword
+	// which indicates the highest priority. Any other name must be defined by
+	// creating a PriorityClass object with that name.
+	// If not specified, the pod priority will be zero.
+	// +optional
+	PriorityClassName string
+	// The priority value which is resolved by the Admission Controller from
+	// PriorityClassName. The higher the value, the higher the priority.
+	// It must not be specified by user.
+	Priority *int32
```

Comment:
Tim Hockin, Jordan and Wojtek raised the concern that setting pod spec should not directly affect status.
This is from Jordan:
> Currently, setting the spec explicitly does not affect the status (and vice versa) (the create/update strategy for all objects with spec preserves the existing status as-is, and the status strategy preserves the existing spec as-is). This changes that behavior for pods, and means that modifying spec can conflict with status updates, and anything that can modify status can change "effective" pod priority.

This is from thockin:

> There's a rule of thumb that everything in status can be recreated from observation. This would not be the case here, as (I assume) priority name->num mapping could change after a pod was created.
To that end, it SHOULD be part of spec, but then you have to answer what happens if a user sets it directly. I would guess you want to disallow that.